action/selection/open_spiel/database.cc

```diff
@@ -483,28 +552,19 @@ std::vector<double> DatabaseState::Returns() const {
 
   // Simulate the game trajectory.
   {
-    // Apply all the tuning actions. This assumes that actions take zero time.
-    for (const auto &player_action : history_) {
-      // Apply the tuning action.
-      {
-        const TuningAction &action = tuner->GetAction(player_action.action);
-        std::string action_sql = action.sql_;
-        // If hypopg is enabled and this is a CREATE INDEX action, we need to modify the action SQL.
-        // TODO(WAN): Be more robust about cReaTE inDeX.
-        bool is_create_index = action_sql.find("CREATE INDEX") == 0 || action_sql.find("create index") == 0;
-        if (use_hypopg && is_create_index) {
-          // hypopg_create_index() is ok with the action_sql ending with a semicolon.
-          action_sql = absl::StrCat("SELECT * FROM hypopg_create_index('", action_sql, "');");
-        }
-        pqxx::result rset{txn.exec(action_sql)};
-      }
-    }
+    ApplyHistory(txn, true /*use_hypopg*/);
 
     // TODO(WAN): Faking out the forecast time should go here.
     //  e.g., pass the current time as an input into GetForecastWorkload(),
     //  and move this block of code into the above for (history) loop.
     //  Right now, the workload only appears at the very end after all actions are applied.
     const std::vector<ForecastedQuery> &workload = forecast->GetForecastWorkload();
+
+    // This vector is used to record predictions. The vector (if valid) corresponds 1:1
```

Comment:
Nit: you are declaring this vector and making people read this, even for code paths that have nothing to do with predictions.

I am OK with this for now, but I think we are approaching the point where we will want to write an `EvaluateWorkloadCost` function that calls `EvaluateWorkloadExplainCost`, `EvaluateWorkloadExplainAnalyzeCost`, etc.

Can you add a comment like

```suggestion
    // TODO(WAN): If you find yourself needing to declare more flag-specific variables here,
    // refactor this into an `EvaluateWorkloadCost` function that calls separate costing functions.
    //
    // This vector is used to record predictions. The vector (if valid) corresponds 1:1
```